lib/AccessControl/View/PoliciesTable/PoliciesTable.test.js

```diff
@@ -70,33 +90,48 @@ describe('PoliciesTable', () => {
         {
           row: 'first',
           rowIndex: 0,
-          expectedData: {
-            ...flattenedPolicies[0],
+          getExpected: () => ({
+            ...sortedByNameAsc[0],
             rowIndex: 0, // MCL adds this information in
-          }
+          }),
         },
         {
           row: 'third',
           rowIndex: 2,
-          expectedData: {
-            ...flattenedPolicies[2],
+          getExpected: () => ({
+            ...sortedByNameAsc[2],
             rowIndex: 2, // MCL adds this information in
-          }
-        }
-      ])('clicking the remove button for the $row row', ({ rowIndex, expectedData }) => {
+          }),
+        },
+      ])('clicking the remove button for the $row row', ({ rowIndex, getExpected }) => {
         beforeEach(async () => {
-          onRemove.mockClear();
+          onDoRemove.mockClear();
           await waitFor(async () => {
             await MultiColumnListCell({ row: rowIndex, columnIndex: 3 }).find(IconButton()).click();
           });
         });
 
         test('onRemove got called with the correct data', async () => {
           await waitFor(() => {
-            expect(onRemove).toHaveBeenCalledWith(expectedData);
+            expect(onDoRemove).toHaveBeenCalledWith(getExpected());
           });
         });
       });
+
+      // NEW: Clicking "Remove" header changes the sort column (branch meta.name !== sortOrder)
```

Comment:
I will pick this up since @CalamityC is off Monday/Tuesday, but this is not our pattern for Jest tests.

Describe/beforeEach does action, then test tests outcome.

This is a bit of an unwieldy pattern I know, but it is important for readability and also for consistency when running in various ci pipelines.


I think about it in a similar vein to some other testing frameworks with a "when X, then Y" pattern. Actions are not the test, you're testing the reaction of the component to the action.

In addition this will throw `act` exceptions, and so we wrap in a `waitFor` to avoid that (Also a cumbersome pattern, but it's better to avoid the errors and be able to cleanly see the test output -- IIRC it also creates issues in CI  if we don't)

Reply:
Hmmm... actually the act errors seem to be gone. I'm going to remove from here for now and hope that those are no longer required... that'd be nice